server/src/main/java/org/eclipse/openvsx/ExtensionService.java

```diff
@@ -141,33 +141,33 @@ private void addDependency(String dependency, ExtensionVersion extVersion) {
         if (split.length != 2 || split[0].isEmpty() || split[1].isEmpty()) {
             throw new ErrorResultException("Invalid 'extensionDependencies' format. Expected: '${namespace}.${name}'");
         }
-        var extension = repositories.findExtension(split[1], split[0]);
-        if (extension == null) {
+        var extensionCount = repositories.countExtensions(split[1], split[0]);
+        if (extensionCount == 0) {
             throw new ErrorResultException("Cannot resolve dependency: " + dependency);
         }
         var depList = extVersion.getDependencies();
         if (depList == null) {
-            depList = new ArrayList<Extension>();
+            depList = new ArrayList<>();
             extVersion.setDependencies(depList);
         }
-        depList.add(extension);
+        depList.add(dependency);
     }
 
     private void addBundledExtension(String bundled, ExtensionVersion extVersion) {
         var split = bundled.split("\\.");
         if (split.length != 2 || split[0].isEmpty() || split[1].isEmpty()) {
             throw new ErrorResultException("Invalid 'extensionPack' format. Expected: '${namespace}.${name}'");
         }
-        var extension = repositories.findExtension(split[1], split[0]);
-        if (extension == null) {
+        var extensionCount = repositories.countExtensions(split[1], split[0]);
+        if (extensionCount == 0) {
```

Comment:
\<out-of-scope>
Here we run into the 1:n database problem, we query the DB for every bundled extension. I don't think that n is big, just saying.
\</out-of-scope>